katcp/core.py

```diff
@@ -299,7 +299,23 @@ def format_argument(self, arg):
         elif isinstance(arg, bool):
             return str(int(arg))
         else:
-            return str(arg)
+            try:
+                return str(arg)
+            except UnicodeEncodeError:
+                # unicode characters will break the str cast, so
+                # try to encode to ascii and replace the offending characters
+                # with a '?' character
+                logger.exception("Error casting message argument to str!")
```

Comment:
Should this also be an exception?

Reply:
I think we can get away with just an error log here because we no longer catch `Exception`